core/data-transformers/orders-transformer.ts

```diff
@@ -0,0 +1,96 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import { getFormatter } from 'next-intl/server';
+
+import { Order } from '@/vibes/soul/sections/order-list-section/order-list-item';
+import { getCustomerOrders } from '~/app/[locale]/(default)/account/orders/page-data';
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+import { ExistingResultType } from '~/client/util';
+
+const ProductAttributes = graphql(`
+  query ProductAttributes($entityIds: [Int!], $first: Int!) {
+    site {
+      products(entityIds: $entityIds, first: $first) {
+        edges {
+          node {
+            entityId
+            path
+          }
+        }
+      }
+    }
+  }
+`);
+
+export const ordersTransformer = async (
+  orders: ExistingResultType<typeof getCustomerOrders>['orders'],
+  format: ExistingResultType<typeof getFormatter>,
+): Promise<Order[]> => {
+  const productIds = orders.reduce<number[]>((acc, order) => {
+    // In the future we probably want to include downloads, email, and pickup consignments
+    const shippingConsignment =
+      order.consignments.shipping?.flatMap((consignment) =>
+        removeEdgesAndNodes(consignment.lineItems),
+      ) ?? [];
+
+    return [
+      ...acc,
+      ...shippingConsignment.map((lineItem) => {
+        return lineItem.productEntityId;
+      }),
+    ];
+  }, []);
+
+  // We need to fetch the product urls separately as our GQL API doesn't have a node
+  // for these urls. In order to make it efficient, we aggregate all the products for
+  // all the orders on the page and fetch them all at once.
+  const { data } = await client.fetch({
+    document: ProductAttributes,
+    // We only display the first 5 orders on the page and we truncate the amount of
+    // products are shown per order. The max products shown per order is 8 based on
+    // the largest screen size, so we can safely fetch 50 products and feel comfortable.
+    // 50 is the max we can fetch at once.
+    variables: { entityIds: productIds, first: 50 },
+    fetchOptions: { next: { revalidate } },
+  });
```

Comment:
This is unfortunate, but the only way I can see us circumventing this is adding a node on our GQL API. Pretty sure the team wanted to avoid adding it though as it may causes an N+1, but I'd need to reach out to the team to confirm that.